rego/example_test.go

```diff
@@ -563,9 +565,66 @@ func ExampleRego_Partial() {
 	for i := range pq.Queries {
 		fmt.Printf("Query #%d: %v\n", i+1, pq.Queries[i])
 	}
+	for i := range pq.Support {
+		fmt.Printf("Support #%d: %v\n", i+1, pq.Support[i])
+	}
 
 	// Output:
 	//
 	// Query #1: "GET" = input.method; input.path = ["reviews", _]; neq(input.is_admin, false)
 	// Query #2: "GET" = input.method; input.path = ["reviews", user3]; user3 = input.user
 }
+
+func ExampleRego_PartialWithOtherUnknowns() {
+
+	ctx := context.Background()
+
+	// Define a simple policy for example purposes.
+	module := `package test
+
+  default allow = false
+
+	allow {
+		input.method = read_methods[_]
+		input.path = ["reviews", user]
+		input.user = user
+	}
+
+	allow {
+		input.method = read_methods[_]
+		input.path = ["reviews", _]
+		input.is_admin
+	}
+
+	read_methods = ["GET"]
+  `
+	input := map[string]interface{}{
+		"method": "GET",
+		"user":   "alice",
+	}
+	input["path"] = []string{"reviews", "alice"}
+
+	r := rego.New(
+		rego.Query("data.test.allow == true"),
```

Comment:
> In this case, OPA is going to have to generate support rules because this query is asking for the value of data.test.allow which implicitly says data.test.allow = x--so we can't omit the default value like I described above.

Thanks, I wasn't aware of that distinction for a moment. However, having changed it, the output goes from
```
Query #1: neq(data.partial.test.allow, false)
Support #1: package partial.test

allow = true { true }
default allow = false
```
to
```
Query #1: equal(data.partial.test.allow, true)
Support #1: package partial.test

allow = true { true }
default allow = false
```

Reply:
Given the current implementation, that's what I would expect in this case--since `allow` has a default rule, OPA blindly generates support rules. I'll file an issue to improve the behaviour for this relatively common case. I think we can improve this for the next release (v0.9) the Compile API.

For now, I would rather leave ExampleRego_Partial as is.